src/Module/DeregisterCommand.php

```diff
@@ -68,8 +69,15 @@ protected function execute(InputInterface $input, OutputInterface $output): int
         $module   = $input->getArgument('module');
         $composer = $input->getOption('composer') ?: 'composer';
 
+        /**
+         * @psalm-suppress InternalClass,InternalMethod We do explicitly need the functionality of the component
+         *                                              installer here
+         */
```

Comment:
This will break though, since we can remove this class at any time in upstream (if it's `@internal`) :thinking:

Reply:
Yah, but I think we can go with this for now since mezzio-tooling and the component-installer are development-only deps and therefore should not break production code.

Not sure if we should address this for now. WDYT?

Reply:
> are development-only deps and therefore should not break production code.

They will still break **within** the organization, which means more maintenance work for us internally.

Reply:
Copy & pasting stuff also creates more maintenance work?
If I now have to copy & paste stuff here, that will take some time as well.

I would rather create a dedicated component which handles the config injection.
Currently, that only works with `regex` and I would love to use `nikic/php-parser` for injection instead.
So if we want to duplicate the logic here, I would prefer extracting all the configuration manipulation logic to a dedicated component.

Reply:
Will copy & paste the logic into this component. extracting it to a dedicated component can still be achieved if we find a bug either in the composer plugin or in this component.

Mantra: Fix problems when there are problems.